runtime/src/main/java/org/corfudb/runtime/collections/CorfuTable.java

```diff
@@ -566,14 +588,22 @@ public void clear() {
     @Override
     @Accessor
     public @Nonnull Set<Entry<K, V>> entrySet() {
-        List<Entry<K, V>> copy = new ArrayList<>(mainMap.size());
-        for (Map.Entry<K, V> entry : mainMap.entrySet()) {
-            copy.add(new AbstractMap.SimpleImmutableEntry<>(entry.getKey(),
-                    entry.getValue()));
-        }
-        return new ImmuableListSetWrapper<>(copy);
+        return mainMap.entryStream().map(entry ->
+                new AbstractMap.SimpleImmutableEntry<>(entry.getKey(), entry.getValue()))
+                .collect(ImmutableSet.toImmutableSet());
     }
 
+    /**
+     * Present the content of a {@link CorfuTable} via the {@link Stream} interface.
+     *
+     * @return stream of entries
+     */
+    @Accessor
+    public @Nonnull Stream<Entry<K, V>> entryStream() {
+        return mainMap.entryStream();
```

Comment:
I believe you need to change the `CheckpointWriter` to consume this API if the underlying `StreamingMap` is using `PersistedStreamingMap`, otherwise the whole RocksDB instance will be recreated in memory.

Reply:
Good catch. I tried changing the `CheckpointWriter` to consume the `StreamingMap`, but the problem is that there is about 20 different tests that need to be modified as a result of this change.

I suppose I will have to address this separately.